lesson_10/solid/solid_app/src/main/java/com/codedifferently/lesson10/library/Library.java

```diff
@@ -156,15 +166,101 @@ public boolean checkInBook(Book book, Patron patron) {
   }
 
   /**
-   * Get the books checked out by a patron.
+   * Get the books currently checked out by a patron.
    *
    * @param patron The patron to get the books for.
-   * @return The books checked out by the patron.
+   * @return The books currently checked out by the patron.
    */
   public Set<Book> getCheckedOutByPatron(Patron patron) {
     return this.checkedOutBooksByPatron.get(patron.getId());
   }
 
+  /**
+   * Check out a book to a librarian.
+   *
+   * @param book The book to check out.
+   * @param librarian The librarian to check out the book to.
+   * @return True if the book was checked out, false otherwise.
+   */
+  public boolean CheckOutMedia(Book item, Librarian librarian) {
```

Comment:
I'd expect that this method would take any `MediaItem` and check it out if possible or return false if the item cannot be checked out.